The `make check-style` done by the Cirrus CI environment could be more helpful

Yet again I pushed a change that resulted in this CI failure:

```
make checkstyle-md
Markdown files that need formatting:
  NEXT-RELEASE.md
Makefile:44: recipe for target 'checkstyle-md' failed
make: *** [checkstyle-md] Error 1

Exit status: 2
```

It would be more useful if a failure actually included the unexpected style violations limited to some reasonable maximum; e.g., 1000 lines.

P.S., The situation that caused me to open this issue involved a single, empty, line I introduced as a consequence of manually resolving a merge conflict. I ran `make style` manually then `git diff` and it reported this:

```
diff --git a/NEXT-RELEASE.md b/NEXT-RELEASE.md
index 7199dd81..a54f97bc 100644
--- a/NEXT-RELEASE.md
+++ b/NEXT-RELEASE.md
@@ -1,7 +1,6 @@
 This is the draft release notes for 0.15.0, scheduled to be released on
 2021-01-01.

-
 # Deprecated features

 The following deprecated features trigger a warning whenever the code is parsed
```

--- a/Makefile
+++ b/Makefile
@@ -33,17 +33,15 @@ style:
 	find . -name '*.go' | xargs goimports -w
 	find . -name '*.md' | xargs prettier --tab-width 4 --prose-wrap always --write
 
+# Check if the style of the Go and Markdown files is correct without modifying
+# those files.
 checkstyle: checkstyle-go checkstyle-md
 
 checkstyle-go:
-	echo 'Go files that need formatting:'
-	! find . -name '*.go' | xargs goimports -l \
-		| sed 's/^/  /' | grep . && echo '  None!'
+	./tools/checkstyle-go.sh
 
 checkstyle-md:
-	echo 'Markdown files that need formatting:'
-	! find . -name '*.md' | xargs prettier --tab-width 4 --prose-wrap always -l \
-		| sed 's/^/  /' | grep . && echo '  None!'
+	./tools/checkstyle-md.sh
 
 .SILENT: checkstyle-go checkstyle-md
 .PHONY: default get generate test style checkstyle checkstyle-go checkstyle-md cover

--- a/tools/checkstyle-go.sh
+++ b/tools/checkstyle-go.sh
@@ -1,0 +1,15 @@
+#!/bin/sh -e
+
+# Check if the style of the Go source files is correct without modifying those
+# files.
+
+# The grep is because `goimports -d` will exit with a non-zero status even
+# when it doesn't suggest any changes. The grep will return failure if there
+# is no output from goimports and sucess if there are any proposed changes.
+
+echo 'Go files need these changes:'
+if find . -name '*.go' | xargs goimports -d | grep .
+then
+    exit 1
+fi
+echo '  None!'

--- a/tools/checkstyle-md.sh
+++ b/tools/checkstyle-md.sh
@@ -1,0 +1,32 @@
+#!/bin/sh -e
+
+# Check if the style of the Markdown files is correct without modifying those
+# files.
+
+# The `prettier` utility doesn't provide a "diff" option. Therefore, we have
+# to do this in a convoluted fashion to get a diff from the current state to
+# what `prettier` considers correct and reporting, via the exit status, that
+# the check failed.
+
+# We predicate the detailed diff on being in a CI environment since we don't
+# care if the files are modified. If not, just list the pathnames that need to
+# be reformatted without actually modifying those files.
+
+if test "$CI" = ""
+then
+    echo 'Markdown files that need changes:'
+    ! find . -name '*.md' |
+        xargs prettier --tab-width 4 --prose-wrap always --list-different |
+        sed 's/^/  /' | grep . && echo '  None!'
+else
+    echo 'Markdown files need these changes:'
+    if ! find . -name '*.md' |
+        xargs prettier --tab-width 4 --prose-wrap always --check >/dev/null
+    then
+        find . -name '*.md' |
+            xargs prettier --tab-width 4 --prose-wrap always --write >/dev/null
+        find . -name '*.md' | xargs git diff
+        exit 1
+    fi
+    echo '  None!'
+fi

